### PR TITLE
Android release fastlane [WIP]

### PR DESCRIPTION
**Pull request by fedecor9**
# :star: Feature

---

#### :pencil2: Description:
                                                        
Android release fastlane setup   _ _ _🚘_ _ _ 

---

### CODE REVIEW

File: .github/workflows/firebase-hosting-pull-request.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# This file was auto-generated by the Firebase CLI
+# https://github.com/firebase/firebase-tools
+
+name: Deploy to Firebase Hosting on PR
```
Comment (mirland) [inline .github/workflows/firebase-hosting-pull-request.yml:4]: shouldn't it be done only on main push?

File: android/app/build.gradle [hunk 2]
```diff
@@ -75,10 +75,10 @@ android {
...
     signingConfigs {
         release {
-            storeFile getEnvVariable('KEYSTORE_FILE') == null ? null : file(getEnvVariable('KEYSTORE_FILE'))
-            storePassword getEnvVariable('KEYSTORE_PASSWORD')
-            keyAlias getEnvVariable('KEYSTORE_KEY_ALIAS')
-            keyPassword getEnvVariable('KEYSTORE_KEY_PASSWORD')
+            storeFile getEnvVariable('RELEASE_STORE_FILE') == null ? null : file(getEnvVariable('RELEASE_STORE_FILE'))
+            storePassword getEnvVariable('RELEASE_STORE_PASSWORD')
+            keyAlias getEnvVariable('RELEASE_KEY_ALIAS')
+            keyPassword getEnvVariable('RELEASE_KEY_PASSWORD')
```
Comment (mirland) [inline android/app/build.gradle:81]: Remember that if you change it, you have to edit bitrise setup

File: android/build/.last_build_id [hunk 3]
```diff
@@ -0,0 +1 @@
+1d88c8556d339bed9217ad90652fa9e3
```
Comment (mirland) [inline android/build/.last_build_id:1]: shouldn't be ignored?

File: android/app/build.gradle [hunk 4]
```diff
@@ -99,3 +99,6 @@ flutter {
 dependencies {
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
 }
+
+android.buildTypes.release.ndk.debugSymbolLevel = 'FULL'
```
Comment (mirland) [inline android/app/build.gradle:103]: use the block structure

File: web_privacy_policy/firebase.json [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+      "**/node_modules/**"
+    ],
+    "rewrites": [
+      {
+        "source": "**",
+        "destination": "/index.html"
+      }
+    ]
+  }
+}
```
Comment (mirland) [inline web_privacy_policy/firebase.json:16]:
```suggestion
}

```

File: web_privacy_policy/public/index.html [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Fluttips</title>
+    <meta http-equiv="refresh" content="0; URL=privacy_policy.html" />
+</head>
+<body style="padding: 5px">
+</body>
+</html>
```
Comment (mirland) [inline web_privacy_policy/public/index.html:12]:
```suggestion
</html>

```

File: web_privacy_policy/public/privacy_policy.html [hunk 7]
```diff
@@ -0,0 +1,147 @@
+<!DOCTYPE html>
```
Comment (mirland) [inline web_privacy_policy/public/privacy_policy.html:1]: can you indent the file please ?

**Review by mirland — APPROVED** [review]

File: android/app/build.gradle (hunk 2 above)
Comment (fedecor9) [reply android/app/build.gradle:81]: Should I make two setups, one for release and another one for testing?

Comment (mirland) [reply android/app/build.gradle:81]: You can use one or two keystore, for now it's the same. My point here is that the current implementation (dev firebase) for example uses `KEYSTORE_FILE`. So you have to update the existing workflows

**Review by SolMendiola — APPROVED** [review]

File: android/app/build.gradle [hunk 8]
```diff
@@ -71,10 +71,20 @@ android {
...
+            storeFile getEnvVariable('RELEASE_STORE_FILE') == null ? null : file(getEnvVariable('RELEASE_STORE_FILE'))
+            storePassword getEnvVariable('RELEASE_STORE_PASSWORD')
+            keyAlias getEnvVariable('RELEASE_KEY_ALIAS')
+            keyPassword getEnvVariable('RELEASE_KEY_PASSWORD')
+        }
+
+        staging {
             storeFile getEnvVariable('KEYSTORE_FILE') == null ? null : file(getEnvVariable('KEYSTORE_FILE'))
             storePassword getEnvVariable('KEYSTORE_PASSWORD')
             keyAlias getEnvVariable('KEYSTORE_KEY_ALIAS')
```
Comment (mirland) [inline android/app/build.gradle:90]: maybe we can release it

File: android/build/.last_build_id (hunk 3 above)
Comment (mirland) [inline android/build/.last_build_id:1]: remove it

File: web_privacy_policy/public/privacy_policy.html [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+  <p>You are advised to review this Privacy Policy periodically for any changes. Changes to this Privacy Policy are
+    effective when they are posted on this page.</p>
+  <h1>Contact Us</h1>
+  <p>If you have any questions about this Privacy Policy, You can contact us:</p>
+  <ul>
+    <li>By email: fluttips@xmartlabs.com</li>
+  </ul>
+</body>
+
+</html>
```
Comment (mirland) [inline web_privacy_policy/public/privacy_policy.html:225]:
```suggestion
</html>

```

File: android/app/build.gradle [hunk 10]
```diff
@@ -99,3 +109,5 @@ flutter {
 dependencies {
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
 }
+
```
Comment (mirland) [inline android/app/build.gradle:112]:
```suggestion
```

**Review by mirland — APPROVED** [review]

**Pull request merged**